app/models/student.rb

```diff
@@ -99,11 +99,9 @@ def absences_overall
   end
 
   def solos(filtered_course = nil)
-    if filtered_course
-      attendance_records.where(pair_ids: [], ignore: nil).where("date between ? and ?", filtered_course.try(:start_date), filtered_course.try(:end_date)).select { |ar| !ar.date.friday? }.count
-    else
-      attendance_records.where(pair_ids: [], ignore: nil).select { |ar| !ar.date.friday? }.count
-    end
+    solo_records = attendance_records.includes(:pairings).where(pairings: {id: nil})
+    filtered_records = filtered_course ? solo_records.where("date between ? and ?", filtered_course.start_date, filtered_course.end_date) : solo_records
+    filtered_records.reject {|ar| ar.date.friday?}.count
```

Comment:
`reject` is a Ruby array method, I believe. This will be more efficient if you write it with AR :)

Reply:
agreed. will work on that.

Reply:
I'm actually not sure if I can write this one as an AR query, because it needs to test whether Date is Friday, which is a Ruby method anyways. Actually I may have a better solution. Attendance records are no longer created on Fridays, and I should just be able to go back and delete all Friday attendance records from the past.

Reply:
Sounds good. Also there are SQL methods for most things :) https://www.w3schools.com/sql/func_mysql_weekday.asp

Reply:
I think I do want to just fix the issue where Friday attendance records are still sometimes being created, but it turns out that means fixing things in a few different places, so I'd like to just leave this until after I merge in this PR, and do it separately, since it's kinda a separate thing.